worker/jobTypes/githubJob.js

```diff
@@ -132,23 +132,51 @@ class GitHubJobClass {
 
     try {
       const exec = workerUtils.getExecPromise();
-      
-      const pullRepoCommands = [
-        `cd repos/${this.getRepoDirName()}`,
-      ];
+      const pullRepoCommands = [`cd repos/${this.getRepoDirName()}`];
 
       // if commit hash is provided, use that
       if (currentJob.payload.newHead) {
-        pullRepoCommands.push(...[
-          `git checkout ${currentJob.payload.newHead}`,
-          `git checkout -b ${currentJob.payload.branchName}`,
-          `git pull origin ${currentJob.payload.branchName}`,
-        ]);
-      } else {
-        pullRepoCommands.push(...[
+        const commitCheckCommands = [
+          `cd repos/${this.getRepoDirName()}`,
+          `git fetch`,
           `git checkout ${currentJob.payload.branchName}`,
-          `git pull origin ${currentJob.payload.branchName}`,
-        ]);
+          `git branch ${currentJob.payload.branchName} --contains ${currentJob.payload.newHead}`
+        ];
+
+        const result = await exec(commitCheckCommands.join('&&'));
+        const returnObject = {};
+        
+        if (!result.stdout.includes(`* ${currentJob.payload.branchName}`)) {
```

Comment:
This means of checking is too brittle, may vary by version of git, by specifics of the relationship to the repository -- i.e., if a commit is in the repo but not that branch, etc. I think we can capture standard error here and whenever we have an error report back to the logger the text of the error -- this will make the slack output more elegant and will capture the full range of errors we might get. The githubPushJob has examples of how to capture standard error from an exec.